sklearn/naive_bayes.py

```diff
@@ -473,7 +474,7 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
             raise ValueError(msg % (X.shape[0], y.shape[0]))
 
         # convert to float to support sample weight consistently
-        Y = Y.astype(np.float64)
+        Y = astype(Y, np.float64, copy=False)
```

Comment:
`label_binarize` always returns `np.int64` values for `Y`. Therefore `Y = astype(Y, np.float64, copy=False)` will never be able to avoid the data conversion and extra malloc. To avoid this we should rather refactor  `label_binarize` to accept an extra `dtype` argument but this should better be done in a dedicated PR.